### ISSUE TITLE
Spike: Define flow in LMS app for Group Provisioning

### ISSUE THREAD

**Issue by lyzadanger** [hypothesis/product-backlog#729]
This is just far enough down the road that I don't want to flesh out too many details lest they become invalidated. But I'd like to spike to define how we want to build this piece of the work.

* This is going to be dependent on #724 or some alternate way to handle the authentication/authorization of this create-group request
* Groups should be created for every _course_ in the target LMS (not per assignment or other more granular datum). This is defined by `context_id` in LTI data.
* Similarly to the user provisioning definition, we'll need to figure out which LTI data map to what in the create-groups request. Hint: from canvas, at least, we do get some nice metadata with course name, etc.
* One approach to flow might involve checking to see if the LMS app already has a record with a `pubid` for a given `context_id` and assuming a group exists if so. This has some potential weaknesses...we should examine.
* Do we need an API service for fetching a single group (bound by authority)? Or is the above logic (check to see if we have a stored `pubid` locally) "good enough" for now?
* The group for a given course needs to be created during a request cycle for an LMS user with an `Instructor` role (or better? I'm not sure, bears discussion). This may lend it more naturally to the "create assignment/document" flow rather than a generic assignment/document launch.
* What's the app's state if a _student_ launches something and there is no associated group yet?

Note: This is a separate task from actually adding uses to existing course groups on their behalf (although the instructor-user who _creates_ the group will get auto-added as member just because that's how `h`'s app logic works).
